### PR TITLE
#174797543 Book an Accommodation 

### PR DESCRIPTION
**Pull request by jpirumvaa**
### What does this PR do?
Allow users to book an accomodation

### Description of Task to be completed?

1. Pull the repo to get the project's latest version
2. Create a branch to work on
3. Select accommodation from the database.
4. Change` isBooked`  to be true and `userId` to be the id of a user who booked accommodation.
5. Validate User's Input
6. Write Tests
7. Add, commit, and push changes to GitHub

### How should this be manually tested?
Clone a repo, run `npm run dev` to run the project. After having a project running on a port using `localhost:PORTNO` to run the project in the browser. Go to `http://localhost:PORTNO/api/accommodation/book` . Make sure that you have `accomodationId` and `roomId` in the request's body. After sending the request, the message of success and the requester's information will be returned.
### Any background context you want to provide?
The request's body should look like this:
`{
 "accommodationId":2,
"roomId": 3,
    "checkInDate": "January 1, 2020",
    "checkOutDate": "Feb 2, 2020"
}`
### What are the relevant pivotal tracker stories?
[#174797543](https://www.pivotaltracker.com/n/projects/2465784)

### CODE REVIEW

File: src/routes/accomodations/bookAccomodation.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+
+const bookAccomodationRoute = new Router()
+
+bookAccomodationRoute.post('/',
+    authMiddleware.checkToken,
+    bookAccomodationValidation,
+    asyncHandler(bookAccomodationController.bookAccomodation))
+
+export default bookAccomodationRoute
```
Comment (b0nbon1) [inline src/routes/accomodations/bookAccomodation.js]: add newline

File: src/routes/accomodations/accomodation.js [hunk 2]
```diff
@@ -4,46 +4,52 @@ import roomController from '../../controllers/rooms';
...
     '/:id/rooms',
-    [checkToken, checkAdmin],
+    //[checkToken, checkAdmin],
+    [checkToken],
     asyncHandler(roomController.getRooms)
   )
   .delete(
     '/:id/rooms',
-    [checkToken, checkAdmin],
+    //[checkToken, checkAdmin],
```
Comment (b0nbon1) [inline src/routes/accomodations/accomodation.js]: remove commented line, if it's not being used

File: src/routes/accomodations/accomodation.js [hunk 3]
```diff
@@ -4,46 +4,52 @@ import roomController from '../../controllers/rooms';
...
+    [checkToken],
     asyncHandler(roomController.makeRooms)
   )
   .get(
     '/:id/rooms',
-    [checkToken, checkAdmin],
+    //[checkToken, checkAdmin],
```
Comment (b0nbon1) [inline src/routes/accomodations/accomodation.js]: remove commented line, if it's not being used

File: src/routes/accomodations/accomodation.js [hunk 4]
```diff
@@ -4,46 +4,52 @@ import roomController from '../../controllers/rooms';
...
     '/:id',
-    [checkToken, checkAdmin],
+    //[checkToken, checkAdmin],
+    [checkToken],
     asyncHandler(accomodationController.updateAccomodation)
   )
   .post(
     '/:id/rooms',
-    [checkToken, checkAdmin],
+    //[checkToken, checkAdmin],
```
Comment (b0nbon1) [inline src/routes/accomodations/accomodation.js]: remove commented line, if it's not being used

File: src/routes/accomodations/accomodation.js [hunk 5]
```diff
@@ -4,46 +4,52 @@ import roomController from '../../controllers/rooms';
...
     '/:id',
-    [checkToken, checkAdmin],
+    //[checkToken, checkAdmin],
+    [checkToken],
     asyncHandler(accomodationController.getAnAccommodation)
   )
   .delete(
     '/:id',
-    [checkToken, checkAdmin],
+    //[checkToken, checkAdmin],
```
Comment (b0nbon1) [inline src/routes/accomodations/accomodation.js]: remove commented line, if it's not being used

File: src/middlewares/validations/bookAccomodation.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+            .messages({
+                'string.empty': res.__("Id of a room cannot be empty"),
+                'any.required': res.__('Id of a room is required')
+            }),
+    })
+
+    const { error } = schema.validate(req.body);
+    if (error) return res.status(400).json({ error: error.details[0].message });
+    return next();
+}
```
Comment (b0nbon1) [inline src/middlewares/validations/bookAccomodation.js]: add newline

File: src/controllers/bookAccomodation.js [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+                email,
+                roomId,
+                bookedAccommodation: accommodationId
+            },
+            selectedAccomodation
+
+        }
+        );
+    }
+}
```
Comment (b0nbon1) [inline src/controllers/bookAccomodation.js]: add newline

**Review by b0nbon1 — CHANGES_REQUESTED** [review]

File: locales/en.json [hunk 8]
```diff
@@ -1,180 +1,188 @@
...
-  "jwt expired": "jwt expired",
-  "Accomodation not found": "Accomodation not found",
-  "No trips found": "No trips found",
-  "Not found": "Not found"
-}
+	"Welcome to barefoot nomad": "Welcome to barefoot nomad",
+	"Email already exists": "Email already exists",
+	"Successfully registered": "Successfully registered",
+	"Email or password is incorrect": "Email or password is incorrect",
+	"Your are successfully loged in": "Your are successfully loged in",
```
Comment (elvisiraguha) [inline locales/en.json:6]: Look out for these spacing changes, and add an empty line at the end of this file.

**Review by dushimeemma — APPROVED** [review]
Smell's good for me but i think you can do CRUD

**Comment by Adadie** [comment]
May you change "accommodation" to "accomodation" or vice versa to keep the format of the codes consistent, also update the endpoint path in the PR description or the readMe file.

**Review by Habinezajanvier — APPROVED** [review]
I like it

**Review by b0nbon1 — APPROVED** [review]
Looks good

**Pull request merged**